server/src/test/java/io/crate/planner/operators/WindowAggTest.java

```diff
@@ -151,6 +159,33 @@ public void test_window_functions_do_not_support_filter_clause() throws Exceptio
             .hasMessage("FILTER is not implemented for non-aggregate window functions (row_number)");
     }
 
+    @Test
+    public void test_partition_by_sub_column_source_plan_output_has_only_parent_in_outputs() throws Exception {
+        LogicalPlan plan = plan("""
+            SELECT ROW_NUMBER() OVER (
+                PARTITION BY obj ['a']
+            ) AS RK
+            FROM (
+                SELECT obj
+                FROM t1
+            ) t
+            """);
+        Collect collect = (Collect) plan.build(
```

Comment:
This test case doesn't cover the fixed code because it hits the `nonDistExecution` branch?

Reply:
Thanks, pushed an update

Reply:
And what about https://github.com/crate/crate/pull/17226 ?